app/src/main/java/com/android/consumerfinancehalan/di/NetworkModule.kt

```diff
@@ -0,0 +1,53 @@
+package com.android.consumerfinancehalan.di
+
+import com.android.consumerfinancehalan.data.service.ApiService
+import com.google.gson.GsonBuilder
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object NetworkModule {
+    private const val BASE_URL = "https://api.halan.io/bff-mobile/vendor-Discovery/public/"
+    @Provides
+    @Singleton
+    @Named("RetrofitObject")
+    fun provideRetrofit(): Retrofit {
+        val gson = GsonBuilder()
```

Comment:
this could have be done better,

Gson Module creation does not have to happen within the network module File